types/src/lib.rs

```diff
@@ -46,6 +46,8 @@ pub struct HotShotConfig<K, ELECTIONCONFIG> {
     pub max_transactions: NonZeroUsize,
     /// List of known node's public keys, including own, sorted by nonce ()
     pub known_nodes: Vec<K>,
+    /// List of known node's public keys and stake value for certificate aggregation, serving as public parameter
+    pub known_nodes_with_stake: Vec<ENTRY>,
```

Comment:
it felt underconstrained w/o specifying the equivalence between the key in `ENTRY` and `K`. 

I wonder if you want to use:
```rust
use hotshot-types::traits::stake_table::StakeTableScheme;

pub struct HotShotConfig<K,ST: StakeTableScheme, ELECTIONCONFIG> 
where ST::Key: K, {
  pub known_nodes: Vec<K>,
  pub stake_table: ST,
}
```

Reply:
1. Yes, that will happen in a separate PR!
2. I agree.  In general HotShot traits have gotten messy.  Here is an issue for this, but we unfortunately have other priorities to work on right now: https://github.com/EspressoSystems/HotShot/issues/1595
3. Normally a PR would have more details!  Sishan's original PR did have more details.  I only made this PR since Sishan can't make it while she is off.

Reply:
Re this particular comment - I agree - good catch.  I think we should probably just remove the known_nodes since it effectively the same thing as known_nodes_with_stake.  Issue here: https://github.com/EspressoSystems/HotShot/issues/1596